include/manif/impl/se_2_3/SE_2_3_base.h

```diff
@@ -0,0 +1,448 @@
+}
+
+template <typename _Derived>
+typename SE_2_3Base<_Derived>::LieGroup
+SE_2_3Base<_Derived>::inverse(OptJacobianRef J_minv_m) const
+{
+  if (J_minv_m)
+  {
+    (*J_minv_m) = -adj();
+  }
+
+  const SO3<Scalar> so3inv = asSO3().inverse();
+
+  return LieGroup(-so3inv.act(translation()),
+                   so3inv,
+                  -so3inv.act(linearVelocity()));
+}
+
+template <typename _Derived>
+typename SE_2_3Base<_Derived>::Tangent
+SE_2_3Base<_Derived>::log(OptJacobianRef J_t_m) const
+{
+  using std::abs;
+  using std::sqrt;
+
+  const SO3Tangent<Scalar> so3tan = asSO3().log();
+
+  Tangent tan((typename Tangent::DataType() <<
+               so3tan.ljac().inverse()*translation(),
+               so3tan.coeffs(),
+               so3tan.ljac().inverse()*linearVelocity()).finished());
+
+  if (J_t_m)
+  {
+    // Jr^-1
+    (*J_t_m) = tan.rjac().inverse();
+  }
+
+  return tan;
+}
+
+template <typename _Derived>
+typename SE_2_3Base<_Derived>::Tangent
+SE_2_3Base<_Derived>::lift(OptJacobianRef J_t_m) const
+{
+  return log(J_t_m);
+}
+
+template <typename _Derived>
+template <typename _DerivedOther>
+typename SE_2_3Base<_Derived>::LieGroup
+SE_2_3Base<_Derived>::compose(
+    const LieGroupBase<_DerivedOther>& m,
+    OptJacobianRef J_mc_ma,
+    OptJacobianRef J_mc_mb) const
+{
+  static_assert(
+    std::is_base_of<SE_2_3Base<_DerivedOther>, _DerivedOther>::value,
+    "Argument does not inherit from SE_2_3Base !");
+
+  const auto& m_se_2_3 = static_cast<const SE_2_3Base<_DerivedOther>&>(m);
+
+  if (J_mc_ma)
+  {
+    (*J_mc_ma) = m.inverse().adj();
+  }
+
+  if (J_mc_mb)
+  {
+    J_mc_mb->setIdentity();
+  }
+
+  return LieGroup(rotation()*m_se_2_3.translation() + translation(),
+                  asSO3().compose(m_se_2_3.asSO3()).quat(),
+                  rotation()*m_se_2_3.linearVelocity() + linearVelocity());
+}
+
+template <typename _Derived>
+template <typename _EigenDerived>
+Eigen::Matrix<typename SE_2_3Base<_Derived>::Scalar, 3, 1>
+SE_2_3Base<_Derived>::act(const Eigen::MatrixBase<_EigenDerived> &v,
+                          tl::optional<Eigen::Ref<Eigen::Matrix<Scalar, 3, 9>>> J_vout_m,
+                          tl::optional<Eigen::Ref<Eigen::Matrix<Scalar, 3, 3>>> J_vout_v) const
+{
+  assert_vector_dim(v, 3);
+
+  const Rotation R(rotation());
+
+  if (J_vout_m)
+  {
+    J_vout_m->template topLeftCorner<3,3>()  =  R;
+    J_vout_m->template block<3,3>(0, 3) = -R * skew(v);
+    J_vout_m->template topRightCorner<3,3>().setZero();
+  }
+
+  if (J_vout_v)
+  {
+    (*J_vout_v) = R;
+  }
+
+  return translation() + R * v;
+}
+
+
+template <typename _Derived>
+typename SE_2_3Base<_Derived>::Jacobian
+SE_2_3Base<_Derived>::adj() const
+{
+  ///
+  /// this is
+  ///       Ad(g) = | R T.R 0|
+  ///               | 0  R  0|
+  ///               | 0 V.R R|
+  ///
+  /// considering vee(log(g)) = (v;w;a)
+  /// with T = [t]_x
+  /// with V = [v]_x
+
+  Jacobian Adj = Jacobian::Zero();
```

Comment:
Not important but this sets the entire matrix to zero and then rewrite half of it. I wonder if setting only the other half to zero would be an improvement.

Reply:
Fixed in https://github.com/artivis/manif/pull/154/commits/810862f30d3e97e9de9dbd54e1a7458146ae91f5